### PR TITLE
 Don't forward all the heaers

### PR DESCRIPTION
**Pull request by sbrunner**
Return message on error
Code style

### CODE REVIEW

File: papyrus_ogcproxy/views.py [hunk 1]
```diff
@@ -29,37 +37,43 @@
...
-                                     body=request.body, headers=h)
+        resp, content = http.request(
+            url, method=request.method, body=request.body, headers=headers
+        )
     except:
         return HTTPBadGateway()
 
     # check for allowed content types
-    if resp.has_key("content-type"):
+    if "content-type" in resp:
```
Comment (elemoine) [inline papyrus_ogcproxy/views.py]: Any particular reason for changing this? Or just stylistic reason?

File: papyrus_ogcproxy/views.py [hunk 2]
```diff
@@ -29,37 +37,43 @@
...
     http = Http(
         disable_ssl_certificate_validation=True,
         proxy_info=proxy_info)
-    h = dict(request.headers)
-    h.pop("Host", h)
+
+    headers = {}
+    for h in forwarded_headers:
+        if h in request.headers:
+            headers[h] = request.headers[h]
```
Comment (elemoine) [inline papyrus_ogcproxy/views.py]: Or more concisely:

```
headers = {h: request.headers[h] for h in forwarded_headers if h in request.headers}
```

File: papyrus_ogcproxy/views.py (hunk 1 above)
Comment (sbrunner) [inline papyrus_ogcproxy/views.py]: Just styling :-)

File: papyrus_ogcproxy/views.py (hunk 2 above)
Comment (sbrunner) [inline papyrus_ogcproxy/views.py]: Why not ...

Comment (elemoine) [inline papyrus_ogcproxy/views.py]: I think it makes sense to use a dict comprehension here. It's more concise, and I don't think it has an impact on readability.

Thanks.

**Comment by elemoine** [comment]
Please fix the spelling in the "Don't forward all the heaers" commit message.

File: papyrus_ogcproxy/views.py (hunk 2 above)
Comment (sbrunner) [inline papyrus_ogcproxy/views.py]: I changed to use dict comprehension, but in multiline to be more readable, is it ok?

**Comment by sbrunner** [comment]
Commit message fixed :-)

File: papyrus_ogcproxy/views.py [hunk 3]
```diff
@@ -29,37 +37,44 @@
...
         disable_ssl_certificate_validation=True,
         proxy_info=proxy_info)
-    h = dict(request.headers)
-    h.pop("Host", h)
+
+    headers = {
+        h: request.headers[h]
+        for h in forwarded_headers
+        if h in request.headers
+    }
```
Comment (elemoine) [inline papyrus_ogcproxy/views.py]: I think you can use fewer lines. I am picky but this formatting looks very unusual to me.

Comment (sbrunner) [inline papyrus_ogcproxy/views.py]: I inline it event I don't found much readable...

**Comment by sbrunner** [comment]
Thanks @elemoine :-)

**Comment by sbrunner** [comment]
It don't seem that I have the rights to upload a package on PyPI, is it possible to creates a new release with this Pull Request?

**Comment by elemoine** [comment]
I can give you the required permissions on pypi I guess. What's your pypi username?

**Comment by sbrunner** [comment]
sbrunner, tanks :-)

**Comment by elemoine** [comment]
> sbrunner, tanks :-)

Done.

**Comment by sbrunner** [comment]
Thanks :-)

**Pull request merged**